### ISSUE TITLE
Unite forces?

### ISSUE THREAD

**Issue by liar666** [xenodium/chatgpt-shell#131]
Hello,

Nice project.

I'm using emacs-aichat for the moment, and I just realized there are many projects in this domain and/or trying to unify the access to any GenerativeAI services. Would be great to unite forces, I'll create & link issues on all the projects so that each of you can think about it :)

= Similar projects (studied a little bit)

    https://github.com/xhcoding/emacs-aichat
    https://github.com/rksm/org-ai
    https://github.com/xenodium/chatgpt-shell
    https://gitlab.com/daanturo/starhugger.el

= Similar projects (just read READMEs)

    https://github.com/emacs-openai/chatgpt
    https://github.com/emacs-openai/codegpt
    https://github.com/CarlQLange/chatgpt-arcana.el
    https://github.com/stuhlmueller/gpt.el
    https://github.com/joshcho/ChatGPT.el
    https://github.com/manateelazycat/mind-wave
    https://github.com/antonhibl/gptai
    https://github.com/samrawal/gpt-emacs-macro

= Project unifying access to local/cloud LLMs (for emacs)

    https://github.com/freckletonj/uniteai

**Comment by xenodium** [comment]
Thanks for getting us all in touch!

It would be nice to unite some of these efforts. Having said that, the current package diversity seems to be working well? By that, I mean, the user experience most of these packages provide differ from one another, and thus cater for different users?

Some packages are org based, this one happens to be shell based, some focus on internals (no external dependencies vs python dependencies), and so on...

When I started this package, there were no shell-based options, so I built one.

Early on, I did get requests to enable different services. I don't have much free time to add different providers, but did spend time in making it easier for others to build other shells via https://xenodium.com/a-shell-maker. I've not seen any usages yet 😅
